src/Workspaces/Core/Portable/Shared/Extensions/ContextQuery/AbstractSyntaxContext.cs

```diff
@@ -87,6 +93,12 @@ private ISet<INamedTypeSymbol> ComputeOuterTypes(CancellationToken cancellationT
             return SpecializedCollections.EmptySet<INamedTypeSymbol>();
         }
 
+        protected static IEnumerable<ITypeSymbol> ComputeInferredTypes(Workspace workspace, SemanticModel semanticModel, int position, CancellationToken cancellationToken)
```

Comment:
I'm still a bit confused.  The only calls to ComputeInferredTypes are in places where we're about to create a SyntaxContext object.   This is a static method *inside* the AbstractSyntaxContext calss.  So why can't the AbstractSyntaxContext just call this itself ?  It is already given the data that this method needs.